internal/providers/terraform/aws/ecs_service.go

```diff
@@ -47,6 +48,88 @@ func NewECSService(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
 	return r.BuildResource()
 }
 
+// calcLaunchType determines the launch type for the resource using the following precedence:
+//   1. aws_ecs_service.launch_type
+//   2. aws_ecs_service.capacity_provider_strategy
+//   3. aws_ecs_service.aws_ecs_cluster.default_capacity_provider_strategy
+//   4. aws_ecs_service.aws_ecs_cluster.aws_ecs_cluster_capacity_providers
+func calcLaunchType(d *schema.ResourceData) string {
+	// Use the launch_type if it is set
+	launchType := d.Get("launch_type").String()
+	if launchType != "" {
+		return launchType
+	}
+
+	// Check for an active direct capacity provider
+	launchType = getCapacityProviderLaunchType(d.Get("capacity_provider_strategy").Array())
+	if launchType != "" {
+		return launchType
+	}
+
+	clusterRefs := d.References("cluster")
+	if len(clusterRefs) > 0 {
+		cluster := clusterRefs[0]
+
+		// check the cluster for a default capacity provider
+
+		if defaultStrategies := cluster.Get("default_capacity_provider_strategy").Array(); len(defaultStrategies) > 0 {
+			launchType = getCapacityProviderLaunchType(defaultStrategies)
+			if launchType == "FARGATE" {
+				return launchType
+			}
+		} else {
+			// since there are no default strategies, look for a directly set fargate capacity provider
+			for _, capProvider := range cluster.Get("capacity_providers").Array() {
+				if capProvider.String() == "FARGATE" {
+					return "FARGATE"
+				}
+			}
+		}
+
+		// check for aws_ecs_cluster_capacity_providers
+		for _, capProvider := range cluster.References("aws_ecs_cluster_capacity_providers.cluster_name") {
+			defaultStrategies := capProvider.Get("default_capacity_provider_strategy").Array()
+			if len(defaultStrategies) > 0 {
+				lt := getCapacityProviderLaunchType(defaultStrategies)
+				if lt == "FARGATE" {
+					return lt
+				}
+				if lt != "" {
+					launchType = lt
+				}
```

Comment:
```suggestion
				if lt != "" {
					return lt
				}
```

Reply:
I think this suggestion isn't right because there can be multiple `aws_ecs_cluster_capacity_providers` and we want to keep looking to see if at least one supports fargate.

Reply:
makes sense